lib/index.js

```diff
@@ -52,140 +57,139 @@ const buildApis = async (apis) => {
+    name: config.get('name'),
+    version: packageJson.version,
+    ignoreTrailingSlash: true,
+    handleUncaughtExceptions: true  //deprecated, but without an equivalent alternative as of restify 8.5.1
+  };
 
-    const extraHeaders = new Set();
-    keys(routes).forEach((routePath) => {
-      routes[routePath].forEach((action) => {
-        (action.expectedHeaders || []).forEach((header) => { extraHeaders.add(header); });
-        debug('routing %s@%s for %s', routePath, action.version, action.name);
-        server[action.method](
-          { path: routePath, version: action.version },
-          cors.headers(),
-          action.handler
-        );
-      });
+  if (config.logRequests) {
+    logger = restify.auditLogger({
+      log: bunyan.createLogger({
+        name: 'audit',
+        stream: process.stdout
+      })
     });
-    server.opts('/*', cors.preflight({ extraHeaders }));
+  }
 
-    server.on('after', function(req, res, route, err) {
-      logger(req, res, route, err);
-    });
-    server.on('NotFound', function(req, res, err, cb) {
-      res.json(404, { type: 'NotFound', message: err.message });
-      cb();
-    });
-    server.on('MethodNotAllowed', function(req, res, err, cb) {
-      res.json(405, { type: 'MethodNotAllowed', message: err.message });
-      cb();
-    });
-    server.on('VersionNotAllowed', function(req, res, err, cb) {
-      res.json(400, { type: 'VersionNotAllowed', message: err.message });
-      cb();
+  server = restify.createServer(serverOptions);
+  server.use(restify.plugins.queryParser({ mapParams: false }));
+  server.use(restify.plugins.gzipResponse());
+  server.use(restify.plugins.bodyParser({ mapParams: false, mapFiles: false }));
+
+  const defs = await buildApis(apis);
+  const routes = {};
+  await forEachSerialP(async (def) => {
+    const api = def[0];
+    const resolved = def[1];
+    api.validate();
+
+    const apiExpectedHeaders = findExpectedHeaders(api.params);
+    const resourcePath = path.resolve(path.dirname(api['x-bravado-sourceFile']), config.get('path'));
+    await forEachSerialP(async (r) => {
+      const resource = api.resources[r];
+      const resourceExpectedHeaders = findExpectedHeaders(resource.params).concat(apiExpectedHeaders);
+      const controllerPath = path.resolve(resourcePath, resource['x-bravado-controller']);
+      await forEachSerialP(async (a) => {
+        const action = resource.actions[a];
+        const actionExpectedHeaders = findExpectedHeaders(action.params).concat(resourceExpectedHeaders);
+
+        let method = action.method.toLowerCase();
+        if (method === 'delete') { method = 'del'; }
+        if (method === 'options') { method = 'opts'; }
+
+        const fullPath = action['x-bravado-fullPath'];
+        const routePath = fullPath.replace(/\{((?:[\w.~-]|(?:%[0-9A-F]{2}))+)\}/g, ':$1');
+        if (!routes[routePath]) { routes[routePath] = []; }
+        let controller = (await import(controllerPath));
```

Comment:
this is the other change I had to make was to turn these three loops into async because import is async unlike require.